Sprint 18

SleeplessShu opened the pull request:
Ещё раз спасибо. Постарался учесть все замечания и внести их.

SleeplessShu committed:
Спринт 18 коммит 00

- Пустой коммит

SleeplessShu committed:
Спринт 18 коммит 01

- Переработана активити Медиатека.
- Добавлены фрагмент Плейлисты
- Добавлен фрагмент Треки

Charmandik commented on app/build.gradle.kts:
стоит вынести в .gitignore

SleeplessShu replied:
сделал

Charmandik commented on app/src/main/res/layout/favorite_playlists_fragment.xml:
```diff
@@ -0,0 +1,53 @@
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_horizontal"
+        android:layout_marginTop="@dimen/dim_24dp"
+        android:paddingTop="@dimen/dim_10dp"
```
Если есть одинаковые android:paddingTop и android:paddingBottom, можно использовать paddingVertical

SleeplessShu replied:
ага, ещё, если они повторяются, то можно в стили их положить.
исправил

Charmandik commented on app/src/main/res/layout/activity_mediateka.xml:
```diff
@@ -1,35 +1,61 @@
-        android:paddingBottom="@dimen/dim_16dp"
-        android:id="@+id/b_back_to_main"
-        android:foreground="?attr/selectableItemBackground">
+        android:paddingTop="@dimen/dim_10dp"
```
В дизайне могут быть нечетные размеры, или не симметричные. Но лучше использовать в коде размеры, кратные 4 или 8. Документация - https://material.io/design/layout/spacing-methods.html#baseline-grid

SleeplessShu replied:
спасибо, я уже получал такой фидбек от ревьюера, но забыл. исправил

Charmandik commented on app/src/main/java/com/practicum/playlistmaker_ver2/mediateka/ui/ActivityMediateka.kt:
```diff
@@ -0,0 +1,48 @@
+class ActivityMediateka : ActivityBase() {
+
+    private val viewModel: FavoritePlaylistsViewModel by viewModel()
+    private lateinit var binding: ActivityMediatekaBinding
```
Вместо использования lateinit, я рекомендую тебе использовать подход с добавлением nullability к типу объекта с оператором ?. и сразу проинициализировать его значением null. Это решение позволяет избежать ошибок, связанных с попытками доступа к неинициализированным объектам, которые могут привести к непредсказуемому поведению или крашам приложения. Хотя это может добавить некоторую дополнительную работу при обращении к объекту через оператор ?. , но это значительно повысит безопасность приложения, так как ты будешь уверен, что объект всегда либо имеет значение, либо равен null, и не будет вызывать исключительных ситуаций.Можешь еще прочитать статью на эту тему https://habr.com/ru/companies/maxilect/articles/590061

SleeplessShu replied:
исправил

Charmandik commented on app/src/main/java/com/practicum/playlistmaker_ver2/mediateka/MediatekaPagerAdapter.kt:
```diff
@@ -0,0 +1,19 @@
+
+class MediatekaPagerAdapter(activity: AppCompatActivity) : FragmentStateAdapter(activity) {
+
+    override fun getItemCount(): Int = 2
```
В Kotlin лучше использовать "return" вместо "=", потому что это помогает делать код более читаемым и явным. Когда ты используешь "return", это явно указывает, что функция возвращает определенное значение. В то же время, использование "=" может вызывать путаницу, особенно если в функции есть несколько выражений.

SleeplessShu replied:
спасибо, исправил

SleeplessShu committed:
Спринт 18 коммит 02

- Фрагменты стартует адаптер.
- Метод переключения темы сделан приватным.
- Некоторые параметры лэйаутов объединены
- Размеры изменены на кратные 4 или 8
- lateinit заменено на nullability
- В адаптере медиатеки вместо = использую return
